cogent/src/Cogent/Glue.hs

```diff
@@ -837,25 +837,57 @@ readEntryFuncs :: [SF.TopLevel TC.DepType TC.TypedPatn TC.TypedExpr]
                -> Last (DS.Typedefs, DS.Constants, [CC.CoreConst CC.UntypedExpr])
                -> M.Map FunName (CC.FunctionType VarName)
                -> [String]
-               -> IO (MN.FunMono VarName)
+               -> IO (Maybe (MN.FunMono VarName))
 readEntryFuncs tced tcState dsState ftypes lns
-  = foldM (\m ln -> readEntryFunc ln >>= \(fn,inst) -> return $ updateFunMono m fn inst) M.empty lns
+  = foldM run (Just M.empty) lns
   where
+    run md ln =
```

Comment:
It can be defined in a monadic way, which might be more concise and more readable.